tests/integration_tests/test_github_integration.py

```diff
@@ -0,0 +1,195 @@
+"""
+Integration tests for GitHub API authentication and PR operations.
+
+These tests verify real GitHub credentials and API functionality.
+They are marked as integration tests and require valid GITHUB_TOKEN.
+"""
+
+import os
+import pytest
+from dotenv import load_dotenv
+from github import Github, Auth
+from coffee_maker.code_formatter.crewai.tools import PostSuggestionToolLangAI
+
+# Load environment variables from .env file
+load_dotenv()
+
+
+@pytest.mark.integration
+class TestGitHubIntegration:
+    """Integration tests for GitHub API operations."""
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        """Verify GITHUB_TOKEN is available."""
+        self.token = os.getenv("GITHUB_TOKEN")
+        if not self.token:
+            pytest.skip("GITHUB_TOKEN environment variable not set")
+
+    def test_github_authentication(self):
+        """Test that we can authenticate with GitHub using current credentials."""
+        auth = Auth.Token(self.token)
+        g = Github(auth=auth)
+
+        # Verify authentication by getting the authenticated user
+        user = g.get_user()
+        assert user is not None
+        assert user.login is not None
+        print(f"✓ Successfully authenticated as: {user.login}")
+
+    def test_github_repo_access(self):
+        """Test that we can access a repository."""
+        auth = Auth.Token(self.token)
+        g = Github(auth=auth)
+
+        # Use the current repository from git status
+        repo_name = "Bobain/MonolithicCoffeeMakerAgent"
+        repo = g.get_repo(repo_name)
+
+        assert repo is not None
+        assert repo.full_name == repo_name
+        print(f"✓ Successfully accessed repository: {repo.full_name}")
+
+    def test_github_pr_access(self):
+        """Test that we can access a pull request."""
+        auth = Auth.Token(self.token)
+        g = Github(auth=auth)
+
+        repo_name = "Bobain/MonolithicCoffeeMakerAgent"
+        repo = g.get_repo(repo_name)
+
+        # Get the list of PRs to find an open one
+        prs = list(repo.get_pulls(state="open"))
+
+        if not prs:
+            pytest.skip("No open pull requests available for testing")
+
```

Comment:
Added a more detailed docstring, including Args and Returns sections using Google style docstrings.
```suggestion
    """Splits a filename into its name and extension.

    Args:
        filename (str): The name of the file.

    Returns:
        Tuple[str, str]: A tuple containing the filename without extension and
            the extension. If no extension is found, the extension will be an
            empty string.
    """
```